python/client.py

```diff
@@ -47,13 +48,15 @@ def save_message(msg: ServiceBusReceivedMessage, download_directory):
         json.dump(raw_json, f)
 
 def get_download_directory(settings: Settings):
-    return settings.RelativeFileDownloadDirectory or 'data'
-
+    if settings.RelativeFileDownloadDirectory and (not settings.RelativeFileDownloadDirectory.isspace()):
```

Comment:
I wonder if we should commit to checking all of these required fields rather than just this one? queue name and others are also really required.

Reply:
A good point - so in the C# .NET client all fields are checked. The python one checks `RelativeFileDownloadDirectory` and then when connecting to the service bus it throws errors if required fields for the connection are missing (set in the Microsoft package, not by us): 
![image](https://user-images.githubusercontent.com/94544807/202504575-7a5dcd54-ad3f-4eb1-82f0-f4c6b80ba0ba.png)

We could throw an error ourselves before it reaches this stage?

For JavaScript we also don't have any checks ourselves for the queue name and the error is less coherent. So perhaps we should throw errors on our side consistently across the clients.

Reply:
Yep, that feels reasonable. For js we can do this quite centrally in `config.js`, and we can do it in python in the `__post_init__` method in `settings.py`

Reply:
> perhaps we should throw errors on our side consistently across the clients

This is something I'm really keen to see, but possibly part of a separate piece of work alongside standardising the files and folder structure of each client.

Reply:
I've refactored the error messages for missing config values across the clients in the latest commit. Now across all clients we have `ServiceBusNamespace`, `ServiceBusQueue` and `RelativeFileDownloadDirectory` as required parameters, that all must have a non-space value. If they are empty / only space characters they will throw the same error message across all clients. This handling now lives in `config.js` for the JavaScript client, in `settings.py` for the Python client and in `Settings.cs` for the C# client.